hyppo/ksample/mean_embedding.py

```diff
@@ -0,0 +1,134 @@
+
+        :math: 'nW_n\Sigma_n^{-1}W_n'
+
+        As seen in the above formulation, this test-statistic takes the same form as
+        the Hotelling :math: 'T^2' statistic. However, the components are
+        defined differently in this case. Given data sets
+        X and Y, define the following as :math: 'Z_i', the vector of differences:
+
+        :math: 'Z_i = (k(X_i, T_1) - k(Y_i, T_1), \ldots, k(X_i, T_J) - k(Y_i, T_J)) \in mathbb{R}^J'
+
+        The above is the vector of differences between kernels at test points, :math: 'T_j'. The kernel
+        maps into the reproducing kernel Hilbert space.
+        This same formulation is used in the Mean Embedding Test.
+        Moving forward, :math: 'W_n' can be defined:
+
+        :math: 'W_n = \frac{1}{n} \sum_{i = 1}^n Z_i
+
+        This leaves :math: '\Sigma_n', the covariance matrix as:
+
+        :math: '\Sigma_n = \frac{1}{n}ZZ^T'
+
+        Once :math: 'S_n' is calculated, a threshold :math: 'r_{\alpha}' corresponding to the
+        :math: '1 - \alpha' quantil of a Chi-squared distribution w/ J degrees of freedom
+        is chosen. Null is rejected if :math: 'S_n' is larger than this threshold.
+
+        References
+        ----------
+        .. footbibliography::
+        """
+    def __init__(self, scale=1, compute_distance=False, bias=False, number_of_random_frequencies=5, **kwargs):
+        self.number_of_frequencies = number_of_random_frequencies
+        self.scale = scale
+        KSampleTest.__init__(
+            self, compute_distance=compute_distance, bias=bias, **kwargs
+        )
+
+    def statistic(self, x, y):
+        _, dimension = np.shape(x)
+        obs = vector_of_differences(dimension, x, y, self.number_of_frequencies, self.scale)
+        return _distance(obs, self.number_of_frequencies)
+
+    def test(self, x, y, random_state=None):
+        r"""
+                Calculates the mean embedding test statistic and p-value.
+
+                Parameters
+                ----------
+                x,y : ndarray
+                    Input data matrices. ``x`` and ``y`` must have the same number of
+                    dimensions. That is, the shapes must be ``(n, p)`` and ``(m, p)`` where
+                    `n` is the number of samples and `p` and `q` are the number of
+                    dimensions.
+
+                Returns
+                -------
+                stat : float
+                    The computed mean embedding statistic.
+                pvalue : float
+                    The computed mean embedding p-value.
+                """
+        check_input = _CheckInputs(
+            inputs=[x,y],
+            indep_test=None
+        )
+        x, y = check_input()
+
+        stat = self.statistic(x,y)
+        pvalue = chi2.sf(stat, self.number_of_frequencies)
+        self.stat = stat
+        self.pvalue = pvalue
+
+        return KSampleTestOutput(stat, pvalue)
+
+
+@jit(nopython=True, cache=True)
+def get_estimate(data, point, scale):
```

Comment:
python convention is the put  _ in front of private functions

Reply:
changed